Code/GraphMol/FileParsers/TDTMolSupplier.cpp

```diff
@@ -89,7 +89,7 @@ TDTMolSupplier::TDTMolSupplier(const std::string &fileName,
   if (!tmpStream || (!(*tmpStream)) || (tmpStream->bad())) {
     std::ostringstream errout;
     errout << "Bad input file " << fileName;
-    if (tmpStream) { delete tmpStream; }
+    delete tmpStream;
```

Comment:
I would point out that "delete" is now a code smell which is pretty nice.

One could do:
```
std::unique_ptr<std::istream*> tmpStream = ...
if (...) {
  std::ostringstream errout;
  errout << "Bad input file " << fileName;
  throw BadFileException(errout.str());
}
dp_inStream = tmpStream.release();
```

It sure is painful that we can't guarantee a move constructor for std::ifstream.  This problem would simply go away.